checkov/cloudformation/graph_builder/variable_rendering/renderer.py

```diff
@@ -264,14 +281,15 @@ def _evaluate_findinmap_connection(value: List[str], dest_vertex_attributes: Dic
         return None, None
 
     @staticmethod
-    def _evaluate_getatt_connection(value: List[str], dest_vertex_attributes: Dict[str, Any]) -> (
-            Optional[str], Optional[str]):
+    def _evaluate_getatt_connection(
+        value: Any, dest_vertex_attributes: Dict[str, Any]
+    ) -> tuple[str | None, str | None]:
         # value = [ "logicalNameOfResource", "attributeName" ]
         try:
             if isinstance(value, list) and len(value) == 2:
                 resource_name = value[0]
                 attribute_at_dest = value[1]
-                dest_name = dest_vertex_attributes.get(CustomAttributes.BLOCK_NAME).split('.')[-1]
+                dest_name = dest_vertex_attributes[CustomAttributes.BLOCK_NAME].split('.')[-1]
```

Comment:
Don't we want to check what happens if we there is no such value in the dict?

Reply:
it was not checked before and would have resulted in a `AttributeError` of `None` having no attribute `split`
if this key is not there, then we have a general problem, which is better reflected through directly accessing it.